local/compose/build.go

```diff
@@ -161,7 +176,37 @@ func (s *composeService) toBuildOptions(service types.ServiceConfig, contextPath
 		Tags:      tags,
 		Target:    service.Build.Target,
 		Exports:   []bclient.ExportEntry{{Type: "image", Attrs: map[string]string{}}},
+		Platforms: platforms,
+	}, nil
+}
+
+func getPlatforms(ps ...string) ([]specs.Platform, error) {
+	var platforms []specs.Platform
+	for _, p := range ps {
+		platform, err := getPlatform(p)
+		if err != nil {
+			return nil, err
+		}
+		platforms = append(platforms, platform)
+	}
+	return platforms, nil
+}
+
+func getPlatform(p string) (specs.Platform, error) {
+	if p == "" {
+		return specs.Platform{
+			OS:           runtime.GOOS,
+			Architecture: runtime.GOARCH,
```

Comment:
For this lines, I've tested on Linux and that works. But something tells me that it doesn't work for MacOS's case, since the CLI runs on the `darwin` and the actual image will run on a `linux`.
@chris-crone Could you confirm that?

Reply:
`runtime.{GOOS,GOARCH}` isn't safe to use on macOS M1 machines as the binary could be being emulated (see: [go-rosetta](https://github.com/tonistiigi/go-rosetta))

We should look at what's done in docker/cli and on moby/moby to see how this is handled